### DISCUSSION TITLE
Upgrade to new version

### DISCUSSION THREAD

**Discussion by karthik-phl** [Infocaster/UrlTracker#6] [Q&A] [answered]
Hi,

If we use the version of Url tracker that's supported for Umbraco 7.15, is it easy/straightforward to upgrade to the latest version when we move to Umbraco 9?

Thanks.

**Comment by D-Inventor** [comment, marked as answer]
Hi there!
In short: No, I don't think there's an easy solution to transfer your old redirects from Umbraco 7 to Umbraco 9.

Here's why:
The URL Tracker stores node id's to redirect to umbraco pages. If you move to Umbraco 9, you will have to create your content from the bottom up again and then the node id's change. The redirects from your Umbraco 7 site will most likely not work at all or even redirect to the wrong page.
Also, from Umbraco 8 and up, the URL Tracker has changed to work with umbraco's multilanguage feature, requiring us to update the database schema. The Umbraco 8 version seems to not be developed to be upgraded from Umbraco 7 (so I have researched, as I've only gotten involved rather recently).
Lastly, the URL Tracker for Umbraco 7 doesn't have the convenient import/export functionality that the Umbraco 8+ version offers.

It will be possible to upgrade from the Umbraco 8 version to the Umbraco 9 version as the database schema for the URL Tracker stays the same and from what I've read, it's possible to migrate an Umbraco 8 website to Umbraco 9

EDIT:
I was just told that in the latest version of the URL Tracker for Umbraco 7, there is an export function, but it's still unsure whether or not it's compatible with the one from Umbraco 8 & 9, though it likely is.

**Reply by karthik-phl** [reply]
Thanks for the response, Dennis.

Hmm, do the node ids change even for an in-place upgrade of Umbraco?

**Reply by D-Inventor** [reply]
If you can do an in-place upgrade from Umbraco 7 to 9, then I don't think you'll have any problems. That being said, I remember Umbraco explicitly stated that they don't support upgrading from 7 to 8. Perhaps you know something that I don't; I assumed that migrating your database from Umbraco 7 to 9 was not possible.

**Reply by karthik-phl** [reply]
I think you're correct re no direct upgrade from 7 to 8; will look into the upgrade path in more detail. Thanks.
